### ISSUE TITLE
Support batch processing of DNN

### ISSUE THREAD

**Issue by beomyeol** [cmssnu/dolphin#141]
In current implementation of `dolphin-dnn`, the neural network model is trained with only one input instance at each time. If we train the model with input batches, we can get better performance on computing activation values especially in fully connected layers since activations of multiple input instances are computed with one matrix multiplication.
Also, in model partitioning, we can split input batches, and each evaluator in the same model takes these splits and computes activation values and gradients.

**Comment by bgchun** [comment]
Good job!
